### PR TITLE
Block Type literals from being used in invalid contexts

### PR DESCRIPTION
**Pull request by adithyaselv**
Today we allow expressions like the following in PowerApps

```
If(Type(Boolean), 1, 2)
``` 

this PR blocks such unintended usage of type literals at compile time.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs [hunk 1]
```diff
@@ -2558,6 +2558,15 @@ public override void Visit(BlankNode node)
...
             public override void Visit(TypeLiteralNode node)
+            {
+                AssertValid();
+                Contracts.AssertValue(node);
+                
+                _txb.SetType(node, DType.Error);
+                _txb.ErrorContainer.Error(node, TexlStrings.ErrTypeLiteral_UnsupportedUsage);
+            }
+
+            private void VisitType(TypeLiteralNode node)
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs:2569]: comment to explain the difference/why the normal visitor pattern isn't right.

File: src/strings/PowerFxResources.en-US.resx [hunk 2]
```diff
@@ -4560,6 +4560,10 @@
     <value>Type literal declaration is invalid. The expression '{0}' cannot be used in a type definition.</value>
     <comment>Error message when validation of type literal fails.</comment>
   </data>
+  <data name="ErrTypeLiteral_UnsupportedUsage" xml:space="preserve">
+    <value>Unsupported usage: Type literals can only be used in type arguments and type definitions.</value>
```
Comment (CarlosFigueiraMSFT) [inline src/strings/PowerFxResources.en-US.resx:4564]: casing: "Unsupported usage: type literals can only be used..."

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**